### PR TITLE
Add garbage collection to README trophies 💪

### PR DESCRIPTION
**Pull request by johanbrandhorst**
Happy to change the naming on the 3 different implementations, but didn't want us to forget to update the README!

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Please let me know if you want me to change the commit message header style, I wasn't sure when it comes to the root of the repo.

File: README.md [hunk 1]
```diff
@@ -57,11 +57,11 @@ Currently supported features:
   * closures
   * bound methods
   * complex numbers (except for arithmetic)
+  * garbage collection (`mark & sweep`, `simple` or `none`)
```
Comment (aykevl) [inline README.md]: 1. I don't know whether we should list `simple` and `none` as garbage collectors, as they are not. Also, at the moment the GC is only implemented on ARM Cortex-M.
2. The rest of the code still calls the `simple` GC the `dumb` gc - this should be consistent.

Comment (johanbrandhorst) [reply README.md]: Okay, if you want we can probably close this until we have GC for all supported platforms (I thought we had a generic implementation when I opened this).

**Comment by deadprogram** [comment]
I think as @johanbrandhorst himself suggested this should be closed until we have a little more clarity on GC per platform. Please reopen or submit a new one as we know better what we are doing.

Thanks!

**Pull request closed without merging**